src/main.rs

```diff
@@ -2,31 +2,39 @@ use std::env;
 use std::process::Command;
 use std::process::exit;
 
-pub fn get_commits(dir: &str, from_revision: &str, to_revision: &str) {
-    let range = format!("{}..{}", from_revision, to_revision);
-
-    let output = Command::new("git")
-            .arg("log")
-            .arg("--oneline")
-            .arg(&range)
-            .current_dir(dir)
-            .output().unwrap_or_else(|e| panic!("Failed to run 'git log' with error: {}", e));
-
-    let stdout_output = String::from_utf8_lossy(&output.stdout);
-    let lines = stdout_output
-            .split('\n')
-            .collect::<Vec<_>>();
-    let mut lines_iterator = lines.iter();
-
-    print!("## {}\n\n", to_revision);
-
-    loop {
-        match lines_iterator.next() {
-            Some(line) => {
-                if line.is_empty() { return; }
-                println!("* {}", line)
-            },
-            None => break
+struct Settings<'a> {
+    repository_path: &'a str,
+    from_revision: &'a str,
+    to_revision: &'a str
+}
+
+impl<'a> Settings<'a> {
```

Comment:
Understandable. 😄 For reading: [Lifetimes](https://doc.rust-lang.org/book/lifetimes.html) in the Rust Book has all the information you are wondering about.